experimental/hip/hip_driver.c

```diff
@@ -0,0 +1,483 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <stdint.h>
+#include <string.h>
+
+#include "experimental/hip/api.h"
+#include "experimental/hip/dynamic_symbols.h"
+#include "experimental/hip/status_util.h"
+#include "iree/base/api.h"
+#include "iree/base/assert.h"
+#include "iree/base/tracing.h"
+#include "iree/hal/api.h"
+
+// Maximum device name length supported by the HIP HAL driver.
+#define IREE_HAL_HIP_MAX_DEVICE_NAME_LENGTH 128
+
+// Utility macros to convert between HIPDevice/hipDevice_t and iree_hal_device_id_t.
```

Comment:
By s/HIPDevice/hipDevice_t/ I meant changing HIPDevice to hipDevice_t. :) Sorry it was following sed synatx. So here it would just be "between hipDevice_t and iree_hal_device_id_t"